src/functions.jl

```diff
@@ -442,70 +460,79 @@ for T in (:Taylor1, :TaylorN)
         @inline function sincos!(s::$T{T}, c::$T{T}, a::$T{T}, k::Int) where {T<:Number}
             if k == 0
                 a0 = constant_term(a)
-                @inbounds s[0], c[0] = sincos( a0 )
+                if $T == Taylor1
+                    @inbounds s[0], c[0] = sincos( a0 )
+                else
+                    @inbounds s[0][1], c[0][1] = sincos( a0 )
+                end
                 return nothing
             end
-            x = a[1]
             zero!(s, k)
             zero!(c, k)
-            if $T == Taylor1
-                @inbounds s[k] = x * c[k-1]
-                @inbounds c[k] = -x * s[k-1]
-            else
-                mul!(s[k], x, c[k-1])
-                mul!(c[k], -x, s[k-1])
-            end
-            @inbounds for i = 2:k
-                x = i * a[i]
+            @inbounds for i = 1:k
                 if $T == Taylor1
+                    x = i * a[i]
                     s[k] += x * c[k-i]
                     c[k] -= x * s[k-i]
                 else
-                    mul!(s[k], x, c[k-i])
-                    mul!(c[k], -x, s[k-i])
+                    mul!(s[k], a[i], c[k-i], scalar=i)
+                    mul!(c[k], a[i], s[k-i], scalar=-i)
                 end
             end
 
-            @inbounds s[k] = s[k] / k
-            @inbounds c[k] = c[k] / k
+            if $T == Taylor1
+                s[k] = s[k] / k
+                c[k] = c[k] / k
+            else
+                @inbounds div!(s[k], s[k], k)
+                @inbounds div!(c[k], c[k], k)
+            end
             return nothing
         end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done 👍